File: packages/shared/src/types/tutor.ts
/** 对话场景 */
export type ConversationScene =
  | 'free_chat'
  | 'homework_help'
  | 'concept_explain'
  | 'review_guide'
  | 'error_analysis';

/** 对话场景标签 */
export const SCENE_LABELS: Record<ConversationScene, string> = {
  free_chat: '自由对话',
  homework_help: '作业辅导',
  concept_explain: '概念讲解',
  review_guide: '复习指导',
  error_analysis: '错题分析',
};

/** 对话 (后端 ConversationOut) */
export interface Conversation {
  id: string;
  student_id: string;
  title: string;
  scene: ConversationScene;
  model_config: Record<string, unknown> | null;
  summary: string | null;
  message_count: number;
  last_message_at: string | null;
  created_at: string;
  updated_at: string;
}

/** 消息 (后端 MessageOut) */
export interface Message {
  id: string;
  conversation_id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  token_count: number | null;
  created_at: string;
}

/** 发送消息响应 */
export interface SendMessageResponse {
  user_msg: Message;
  assistant_msg: Message;
}

/** 反馈请求 */
export interface FeedbackRequest {
  message_id: string;
  rating: number;
  comment?: string;
}

File: services/interfaces/schemas/tutor.py
"""
AI Tutor Pydantic schemas
──────────────────────────
会话 & 消息的请求 / 响应数据结构
"""

from __future__ import annotations

from datetime import datetime
from uuid import UUID
from typing import Optional

from pydantic import BaseModel, Field

from interfaces.schemas.base import OrmBase


# ── Conversation ──────────────────────────────────────


class ConversationCreate(BaseModel):
    """创建会话请求"""

    scene: str = Field(
        default="free_chat",
        description="场景: free_chat|homework_help|concept_explain|review_guide|error_analysis",
    )
    title: Optional[str] = Field(None, max_length=200)
    context: Optional[dict] = Field(
        None,
        description="上下文: task_id, chapter_id, knowledge_point_ids, difficulty, student_grade",
    )


class ConversationUpdate(BaseModel):
    """更新会话请求（部分更新）"""

    title: Optional[str] = Field(None, max_length=200)
    status: Optional[str] = Field(None, description="active|archived")


class ConversationOut(OrmBase):
    """会话响应"""

    id: UUID
    student_id: UUID
    title: Optional[str]
    scene: str
    status: str
    message_count: int
    last_message_at: Optional[datetime]
    context: Optional[dict]
    created_at: datetime
    updated_at: datetime


# ── Message ───────────────────────────────────────────


class MessageSend(BaseModel):
    """发送消息请求"""

    content: str = Field(..., min_length=1, max_length=5000, description="用户消息")


class MessageOut(OrmBase):
    """消息响应"""

    id: UUID
    conversation_id: UUID
    role: str
    content: str
    token_count: Optional[int]
    model_name: Optional[str]
    created_at: datetime


# ── Feedback ──────────────────────────────────────────


class FeedbackCreate(BaseModel):
    """消息反馈请求"""

    rating: int = Field(..., ge=1, le=5, description="评分 1-5")
    comment: Optional[str] = Field(None, max_length=500)
